src/sinks/honeycomb/sink.rs

```diff
@@ -0,0 +1,78 @@
+//! Implementation of the `honeycomb` sink.
+
+use crate::sinks::{
+    prelude::*,
+    util::http::{HttpJsonBatchSizer, HttpRequest},
+};
+
+use super::request_builder::HoneycombRequestBuilder;
+
+pub(super) struct HoneycombSink<S> {
+    service: S,
+    batch_settings: BatcherSettings,
+    request_builder: HoneycombRequestBuilder,
+}
+
+impl<S> HoneycombSink<S>
+where
+    S: Service<HttpRequest> + Send + 'static,
+    S::Future: Send + 'static,
+    S::Response: DriverResponse + Send + 'static,
+    S::Error: std::fmt::Debug + Into<crate::Error> + Send,
+{
+    /// Creates a new `HoneycombSink`.
+    pub(super) const fn new(
+        service: S,
+        batch_settings: BatcherSettings,
+        request_builder: HoneycombRequestBuilder,
+    ) -> Self {
+        Self {
+            service,
+            batch_settings,
+            request_builder,
+        }
+    }
+
+    async fn run_inner(self: Box<Self>, input: BoxStream<'_, Event>) -> Result<(), ()> {
+        let service = ServiceBuilder::new().service(self.service);
```

Comment:
I believe this line is essentially a noop.
```suggestion
```

Reply:
ah yep. forgot that was carry over from the http sink